src/linker/Linker/MethodReferenceExtensions.cs

```diff
@@ -10,6 +10,7 @@ namespace Mono.Linker
 {
 	public static class MethodReferenceExtensions
 	{
+		[System.Diagnostics.CodeAnalysis.SuppressMessage ("ApiDesign", "RS0030:Do not used banned APIs", Justification = "MethodReference.Resolve is banned for performance reasons. GetDisplayName should be a really cold path and shouldn't require a context to call.")]
```

Comment:
- Please use pragma
- Would it be too ugly to pass the context to GetDisplayName?

Reply:
It might not necessarily be ugly, but I think it would take significant effort to bubble up the context parameter to GetDisplayName on all the SingleValue types and TypeSystemProxy types, and it's really only called from diagnostics which should be a very cold path anyway.